cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -502,14 +515,37 @@ instance IsShelleyBasedEra era => FromJSON (TxOut CtxTx era) where
 
             mReferenceScript <- o .:? "referenceScript"
 
-            reconcile alonzoTxOutInBabbage mInlineDatum mReferenceScript
+            reconcileBabbage alonzoTxOutInBabbage mInlineDatum mReferenceScript
+
+          ShelleyBasedEraConway -> do
+            alonzoTxOutInConway <- alonzoTxOutParser ScriptDataInConwayEra o
+
+            -- We check for the existence of inline datums
+            inlineDatumHash <- o .:? "inlineDatumhash"
+            inlineDatum <- o .:? "inlineDatum"
+            mInlineDatum <-
+              case (inlineDatum, inlineDatumHash) of
+                (Just dVal, Just h) ->
+                  case scriptDataFromJson ScriptDataJsonDetailedSchema dVal of
+                    Left err ->
+                      fail $ "Error parsing TxOut JSON: " <> displayError err
+                    Right sData ->
+                      if hashScriptDataBytes sData /= h
+                      then fail "Inline datum not equivalent to inline datum hash"
+                      else return $ TxOutDatumInline ReferenceTxInsScriptsInlineDatumsInConwayEra sData
+                (Nothing, Nothing) -> return TxOutDatumNone
+                (_,_) -> fail "Should not be possible to create a tx output with either an inline datum hash or an inline datum"
+
+            mReferenceScript <- o .:? "referenceScript"
+
+            reconcileConway alonzoTxOutInConway mInlineDatum mReferenceScript
          where
-           reconcile
+           reconcileBabbage
              :: TxOut CtxTx BabbageEra -- ^ Alonzo era datum in Babbage era
              -> TxOutDatum CtxTx BabbageEra -- ^ Babbagae inline datum
```

Comment:
Can you fix this typo while you're here?  `s/Babbagae/Babbage`

Also all other occurrences in this file.

Reply:
Done!